ledger/participant-integration-api/src/main/scala/platform/store/CompletionFromTransaction.scala

```diff
@@ -36,30 +37,23 @@ private[platform] object CompletionFromTransaction {
   // * -Xfatal-warnings _MUST_ be enabled
   def toErrorCode(rejection: RejectionReason): Code = {
     rejection match {
-      case _: RejectionReason.Inconsistent | _: RejectionReason.Disputed |
-          _: RejectionReason.PartyNotKnownOnLedger =>
-        Code.INVALID_ARGUMENT
-      case _: RejectionReason.ResourcesExhausted | _: RejectionReason.InvalidLedgerTime =>
+      case _: Disputed | _: PartyNotKnownOnLedger => Code.INVALID_ARGUMENT
+      case _: Inconsistent | _: ResourcesExhausted | _: InvalidLedgerTime => {
         Code.ABORTED
-      case _: RejectionReason.SubmitterCannotActViaParticipant =>
-        Code.PERMISSION_DENIED
+      }
+      case _: SubmitterCannotActViaParticipant => Code.PERMISSION_DENIED
     }
   }
 
-  private def toParticipantRejection(reason: domain.RejectionReason): RejectionReason =
+  def toParticipantRejection(reason: domain.RejectionReason): RejectionReason =
```

Comment:
Please move this somewhere else if you're going to reuse it.

Reply:
Moved to `Conversions`